scripts/run_lighthouse_a11y_tests.py

```diff
@@ -0,0 +1,174 @@
+
+    except subprocess.CalledProcessError:
+        python_utils.PRINT(
+            'Puppeteer script failed. More details can be found above.')
+        sys.exit(1)
+
+
+def export_url(url):
+    """Exports the url to an environmental variable."""
+    url_list = url.split('/')
+    if 'collection_editor' in url:
+        os.environ['collection_editor'] = url_list[5]
+    elif 'create' in url:
+        os.environ['exploration_editor'] = url_list[4]
+    elif 'topic_editor' in url:
+        os.environ['topic_editor'] = url_list[4]
+    elif 'story_editor' in url:
+        os.environ['story_editor'] = url_list[4]
+    elif 'skill_editor' in url:
+        os.environ['skill_editor'] = url_list[4]
+    else:
+        return
+
+
+def run_lighthouse_checks():
+    """Runs the lighthouse checks through the .lighthouserc.js config."""
+
+    node_path = os.path.join(common.NODE_PATH, 'bin', 'node')
+    lhci_path = os.path.join('node_modules', '@lhci', 'cli', 'src', 'cli.js')
+    bash_command = [
+        node_path, lhci_path, 'autorun', '--config=.lighthouserc-a11y.js']
+
+    try:
+        subprocess.check_call(bash_command)
+        python_utils.PRINT('Lighthouse checks completed successfully.')
+    except subprocess.CalledProcessError:
+        python_utils.PRINT(
+            'Lighthouse checks failed. More details can be found above.')
+        sys.exit(1)
+
+
+def enable_webpages():
+    """Enables deactivated webpages for testing."""
+
+    pattern = 'CONTRIBUTOR_DASHBOARD_ENABLED = .*'
+    replace = 'CONTRIBUTOR_DASHBOARD_ENABLED = True'
+    common.inplace_replace_file(FECONF_FILE_PATH, pattern, replace)
+
+    pattern = '"ENABLE_ACCOUNT_DELETION": .*'
+    replace = '"ENABLE_ACCOUNT_DELETION": true,'
+    common.inplace_replace_file(CONSTANTS_FILE_PATH, pattern, replace)
+
+
+def start_google_app_engine_server():
+    """Start the Google App Engine server."""
+
+    app_yaml_filepath = 'app_dev.yaml'
+    p = subprocess.Popen(
+        '%s %s/dev_appserver.py --host 0.0.0.0 --port %s '
+        '--clear_datastore=yes --dev_appserver_log_level=critical '
+        '--log_level=critical --skip_sdk_update_check=true %s' %
+        (
+            common.CURRENT_PYTHON_BIN, common.GOOGLE_APP_ENGINE_SDK_HOME,
+            GOOGLE_APP_ENGINE_PORT, app_yaml_filepath
+        ), shell=True)
+    SUBPROCESSES.append(p)
+
+
+def main():
+    """Runs lighthouse checks and deletes reports."""
+
+    enable_webpages()
+    atexit.register(cleanup)
+
+    build.main(args=[])
+    background_processes = []
+    python_utils.PRINT('Compiling webpack...')
+    webpack_config_file = build.WEBPACK_DEV_CONFIG
+    background_processes.append(subprocess.Popen([
+        common.NODE_BIN_PATH,
+        os.path.join(
+            common.NODE_MODULES_PATH, 'webpack', 'bin', 'webpack.js'),
+        '--config', webpack_config_file, '--watch']))
+
+    # Give webpack few seconds to do the initial compilation.
+    time.sleep(10)
```

Comment:
instead of waiting for a fixed amount of time can we wait for webpack compilation to complete. See how it's done in run_e2e_tests.py file if required you can move that function to common.py and re-use here!

Reply:
Done! Although I am getting this socket error, but it doesn't stop me from running the tests. Any ideas of what could cause it?

```
socket.error 41
Traceback (most recent call last):
  File "/Users/jimzhan/opensource/oppia_tools/google-cloud-sdk-304.0.0/google-cloud-sdk/platform/google_appengine/lib/cherrypy/cherrypy/wsgiserver/wsgiserver2.py", line 1302, in communicate
    req.respond()
  File "/Users/jimzhan/opensource/oppia_tools/google-cloud-sdk-304.0.0/google-cloud-sdk/platform/google_appengine/lib/cherrypy/cherrypy/wsgiserver/wsgiserver2.py", line 831, in respond
    self.server.gateway(self).respond()
  File "/Users/jimzhan/opensource/oppia_tools/google-cloud-sdk-304.0.0/google-cloud-sdk/platform/google_appengine/lib/cherrypy/cherrypy/wsgiserver/wsgiserver2.py", line 2127, in respond
    self.write(chunk)
  File "/Users/jimzhan/opensource/oppia_tools/google-cloud-sdk-304.0.0/google-cloud-sdk/platform/google_appengine/lib/cherrypy/cherrypy/wsgiserver/wsgiserver2.py", line 2188, in write
    self.req.write(chunk)
  File "/Users/jimzhan/opensource/oppia_tools/google-cloud-sdk-304.0.0/google-cloud-sdk/platform/google_appengine/lib/cherrypy/cherrypy/wsgiserver/wsgiserver2.py", line 879, in write
    self.conn.wfile.sendall(chunk)
  File "/Users/jimzhan/opensource/oppia_tools/google-cloud-sdk-304.0.0/google-cloud-sdk/platform/google_appengine/lib/cherrypy/cherrypy/wsgiserver/wsgiserver2.py", line 970, in sendall
    bytes_sent = self.send(data)
  File "/Users/jimzhan/opensource/oppia_tools/google-cloud-sdk-304.0.0/google-cloud-sdk/platform/google_appengine/lib/cherrypy/cherrypy/wsgiserver/wsgiserver2.py", line 977, in send
    bytes_sent = self._sock.send(data)
error: [Errno 41] Protocol wrong type for socket
```